File: design-patterns/go/command/ui_element/ui_control.go
// ui_control.go

package main

type UiControl struct {
	commandList []UiCommand
}

func NewUiControl() (uiControl *UiControl) {
	uiControl = &UiControl{}
	// uiControl.commandList = make([]UiCommand)
	return
}

func (uiControl *UiControl) AddElement(uiCommand UiCommand) {
	// Execute command
	uiCommand.Print()

	// Add to list
	uiControl.commandList = append(uiControl.commandList, uiCommand)
}

func (uiControl *UiControl) RemoveElement(uiCommand UiCommand) {
	// Remove element
	uiCommand.Remove()

	// Remove from list
	newList := []UiCommand{}

	for _, elem := range uiControl.commandList {
		if elem != uiCommand {
			newList = append(newList, elem)
		}
	}

	uiControl.commandList = newList
}

func (uiControl *UiControl) Undo() {
	uiCommand := uiControl.commandList[len(uiControl.commandList) - 1]
	uiControl.RemoveElement(uiCommand)
}

File: design-patterns/php/command/uielement/UiControl.php
<?php
// UiControl.php

namespace BigBoxCode\DesignPattern\Command\UiElement;


class UiControl {
    private array $commandList = [];

    public function addElement(UiCommand $command) {
        $command->print();

        array_push($this->commandList, $command);
    }

    public function removeElement(UiCommand $command) {
        $command->remove();

        foreach ($this->commandList as $key => $currentCommand) {
            if ($command === $currentCommand) {
                unset($this->commandList[$key]);
                break;
            }
        }
    }

    // Remove the last command
    public function undo() {
        $command = end($this->commandList);

        $this->removeElement($command);
    }
}